### ISSUE TITLE
[Appearance] Alignment of inlay name hints in multi-line struct and arglist

### ISSUE THREAD

**Issue by zhangyx1998** [clangd/vscode-clangd#719]
Is it possible to "ljust" names of these inlay hints to the max width among them so they look better in the editor? (i.e. all `=` on the same column).

An realistic case where it will definitely help:

<img width="735" alt="image" src="https://github.com/user-attachments/assets/b6ece0ed-9a0e-4c7d-8f7f-82b10c073ed5">

Notice that the comments were intended to be aligned:

<img width="732" alt="image" src="https://github.com/user-attachments/assets/3f45defc-139f-4a0e-8774-6be3c18d53f7">

**Comment by HighCommander4** [comment]
Do you have in mind a mechanism for how this would work? Would the language server pad the text contents of the hints with spaces, and assume the client uses a monospace font so that this results in visual alignment?

**Comment by zhangyx1998** [comment]
I don't think this is the responsibility of backend (clangd). To me this feature is better to be implemented in the plugin - it's UI stuff.

**Comment by HighCommander4** [comment]
The vscode-clangd plugin doesn't really have any involvement in the rendering of inlay hints. The requests to the server are sent by the [vscode-languageclient](https://github.com/microsoft/vscode-languageserver-node/tree/main/client) library (which is language agnostic), and the rendering of the resulting hints is handled by vscode itself.

So, I don't know how this would work.

**Comment by zhangyx1998** [comment]
Interesting. My assumption was to find a place for this feature in [src/inlay-hints.ts](https://github.com/clangd/vscode-clangd/blob/master/src/inlay-hints.ts). However the comment in that file says it will eventually be removed.

Will it make any sense to insert white spaces on language server side (as a fallback solution)?

**Comment by HighCommander4** [comment]
> My assumption was to find a place for this feature in [src/inlay-hints.ts](https://github.com/clangd/vscode-clangd/blob/master/src/inlay-hints.ts). However the comment in that file says it will eventually be removed.

Yeah, that's an old implementation of inlay hints before it was standardized in the Language Server Protocol.

> Will it make any sense to insert white spaces on language server side (as a fallback solution)?

It would have to be opt-in through a setting, since we can't assume the client uses a monospace font in general, and it would still be a non-trivial effort to recognize the places/situations where this kind of alignment should be performed, but it seems feasible in principle.

I'll move this into to the server repo for now.
